backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py

```
@@ -0,0 +1,225 @@
+    # Use upload_files_async pattern for proper file handling with cleanup
+    zip_file_handle = None
+    try:
+        # Read file content
+        file_content = await file.read()
+        
+        # Save zip file using proper file handling pattern from upload_files_async
+        try:
+            zip_file_handle = open(zip_path, 'wb')
+            zip_file_handle.write(file_content)
+        except IOError as e:
+            logger.error(f"Error writing zip file {zip_path}: {e}")
```

Ruff (0.14.8)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)